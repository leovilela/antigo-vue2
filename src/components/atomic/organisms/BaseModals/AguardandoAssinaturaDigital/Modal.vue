<template>
  <Base-Modal v-if="active" class="assinatura-digital">
    <template #header>
      <img src="~/assets/img/logo-listo-smartphone.png" alt="" height="100px" />
    </template>
    <template #body>
      <h3>Assinatura digital</h3>
      <div class="documentos">
        <p>Informe seu cliente que é necessário assinar digitalmente os seguintes documentos:</p>
        <p>- CCB (Cédula de Crédito Bancário);</p>
        <p>- CET (Custo Efetivo Total); e</p>
        <p>- Ficha Cadastral.</p>
        <p>Estes documentos podem ser acessados na página inicial do aplicativo.</p>
      </div>
      <div class="smartphone" v-show="isSmartphone">
        <img src="~/assets/img/modal-assinatura-digital-smartphone.png" alt="" height="527px" />
      </div>
    </template>
    <template #footer>
      <Base-Button
        class="btn-mostrar-exemplo"
        aria-label="Mostrar Exemplo"
        @click="mostrarExemplo()"
      >
        {{ buttonText }}
      </Base-Button>
      <Base-Button
        class="btn-continuar"
        aria-label="Continuar"
        @click="$emit('close')"
      >
        CONTINUAR
      </Base-Button>
    </template>
  </Base-Modal>
</template>

<script>
export default {
  data() {
    return {
      isSmartphone: false,
    };
  },
  props: {
    active: Boolean,
  },
  methods: {
    mostrarExemplo() {
      this.isSmartphone = !this.isSmartphone;
    },
  },
  computed: {
    buttonText() {
      return this.isSmartphone ? "ESCONDER EXEMPLO" : "MOSTRAR EXEMPLO";
    },
  },
};
</script>

<style lang="less">
@import "~assets/vars.less";
  .assinatura-digital {
  width: 70%;
  .modal {
    width: 70%;
    position: relative;
    overflow: visible;
  }
  h3 {
    font-family: 'newjunebold';
    font-size: 30px;
    color: @color1;
    margin: 30px 0 20px 0;
    text-align: center;
  }
  h4 {
    font-family: 'newjuneregular';
    font-size: 16px;
    color: @color1;
    font-weight: bold;
    margin: 0;
    line-height: 14px;
  }
  ul{
    list-style-type: none;
    padding: 0;
  }
  p {
    font-family: 'newjuneregular';
    font-size: 16px;
    color: @color1;
    text-align: left;
  }
  .documentos {
    text-align: left;
  }
  .smartphone {
    background: #fff;
    padding: 20px;
    position: absolute;
    top: -140px;
    right: -310px;
    z-index: 9999;
    @media screen and (min-width: 1920px){
      padding: 18.5px;
    }
  }
  .btn-continuar {
    font-family: 'newjunebold';
    font-size: 20px;
    width: auto;
    background: #000;
    color: #fff;
    margin-top: 40px;
    padding: 0 60px;
    text-transform: uppercase;
  }
  .btn-mostrar-exemplo {
    font-family: 'newjunebold';
    font-size: 20px;
    width: auto;
    background: #fff;
    color: #000;
    margin-top: 40px;
    padding: 0 20px;
    text-transform: uppercase;
    border: 1px solid #000;
  }
}

</style>
