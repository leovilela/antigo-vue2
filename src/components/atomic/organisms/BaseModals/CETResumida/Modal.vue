<template>
  <Base-Modal v-if="active" class="cet">
    <template #header>
      <img src="~/assets/img/logo-carro-custo.png" alt="" />
      <Base-Button
        aria-label="Close modal"
        class="area-chassi__close"
        @click="closeModal"
        >x</Base-Button
      >
    </template>
    <template #body>
      <h3>CET - Resumida</h3>
      <h4>Custo Efetivo Total</h4>
      <ul>
        <li>
          <p>
            <strong>Valor do bem:</strong>
            {{ cet.shortCet.value | convertMoney }}
          </p>
        </li>
        <li>
          <p>
            <strong>Entrada:</strong>
            {{ cet.shortCet.inputValue | convertMoney }}
          </p>
        </li>
        <li>
          <p><strong>Marca:</strong> {{ cet.shortCet.brand }}</p>
        </li>
        <li>
          <p><strong>Modelo:</strong> {{ cet.shortCet.model }}</p>
        </li>
        <li>
          <p><strong>Cor:</strong> {{ cet.shortCet.color }}</p>
        </li>
        <li>
          <p>
            <strong>Ano de fabricação:</strong> {{ cet.shortCet.modelYear }}
          </p>
        </li>
        <li>
          <p><strong>Placa:</strong> {{ cet.shortCet.licensePlate }}</p>
        </li>
        <li>
          <p>
            <strong>Data do primeiro Vencimento:</strong>
            {{ cet.shortCet.firstDate }}
          </p>
        </li>
        <li>
          <p>
            <strong>Data do último vencimento:</strong>
            {{ cet.shortCet.lastDate }}
          </p>
        </li>
        <li>
          <p>
            <strong>Valor do IOF:</strong>
            {{ cet.shortCet.iofValue | convertMoney }}
          </p>
        </li>
        <li>
          <p><strong>CET ao mês:</strong> {{ cet.shortCet.cetMonth }} %</p>
        </li>
        <li>
          <p><strong>CET ao ano:</strong> {{ cet.shortCet.cetYear }} %</p>
        </li>
        <li>
          <p>
            <strong>Taxa de juros efetiva ao mês:</strong>
            {{ cet.shortCet.interestRateMonth }} %
          </p>
        </li>
        <li>
          <p>
            <strong>Valor da TC3:</strong>
            {{ cet.shortCet.tc3Value | convertMoney }}
          </p>
        </li>
        <li>
          <p>
            <strong>Taxa de avaliação do veículo:</strong>
            {{ cet.shortCet.vehicleEvaluationFee | convertMoney }}
          </p>
        </li>
      </ul>
    </template>
    <template #footer>
      <section class="area-chassi">
        <Base-FormGroup
          v-if="form"
          type="text"
          name="chassiNumber"
          :label="$t('labels.veiculo.numero_chassi')"
          v-model.trim="$v.chassi.$model"
          @input="delayTouch($v.chassi)"
          @keydown.enter="payNow"
          :class="{ 'area-chassi--error': $v.chassi.$error }"
          class="area-chassi__information"
        >
          <span
            class="area-chassi__error-mesage"
            v-if="$v.chassi.$error && !$v.chassi.required"
          >
            {{ $t("error.chassi_obrigatorio") }}
          </span>
          <span
            class="area-chassi__error-mesage"
            v-if="
              $v.chassi.$error && $v.chassi.required && !$v.chassi.minLength
            "
          >
            {{ $t("error.chassi_min_length") }}
          </span>
          <span
            class="area-chassi__error-mesage"
            v-if="
              $v.chassi.$error &&
                !$v.chassi.validateChassi &&
                $v.chassi.required &&
                $v.chassi.minLength
            "
          >
            {{ $t("error.chassi_invalido") }}
          </span>
        </Base-FormGroup>

        <div class="area-chassi__buttons">
          <Base-Button
            class="area-chassi__button area-chassi__button--close"
            aria-label="Close modal"
            type="button"
            @click="closeCET"
            >PAGAR DEPOIS</Base-Button
          >

          <Base-Button
            aria-label="Close modal"
            :disable="$v.chassi.$invalid && form"
            :class="inputClass"
            @click="payNow"

            >PAGAR AGORA</Base-Button
          >
        </div>
      </section>
    </template>
  </Base-Modal>
</template>

<script>
import { required, minLength } from "vuelidate/lib/validators";
import { mapMutations, mapState, mapActions } from "vuex";
import { CLOSE_SIMULATION } from "~/store/simulacao/mutations";
import { validateChassi } from "~/utils/validate.utils";

const touchMap = new WeakMap();

export default {
  filters: {
    convertMoney(value) {
      return `R$ ${value}`;
    },
  },
  validations: {
    chassi: {
      required,
      minLength: minLength(17),
      validateChassi,
    },
  },
  data() {
    return {
      chassi: '',
      form: false,
      button: false,
    };
  },
  props: {
    cet: {
      type: Object,
      required: true,
    },
    active: {
      type: Boolean,
      required: false,
      default: false,
    },
    affiliateId: {
      type: Number,
      required: true,
    },
    registerLoanId: {
      type: Number,
      default: 0,
    },
    zeroKm: {
      type: Boolean,
      default: false,
    },
    hasRescueOperation: {
      type: Boolean,
      default: false,
    },
  },
  methods: {
    ...mapMutations({
      closeSimulation: `simulacao/${CLOSE_SIMULATION}`,
      startLoad: "startLoading",
      finishLoad: "finishLoading",
    }),
    ...mapActions({
      createChassi: "simulacao/fipe/createChassi",
    }),
    async closeCET() {
      try {
        if (!this.hasRescueOperation) {
          await this.finishLater();
          return;
        }

        this.closeModal();
      } catch (error) {
        console.error(error);
      } finally {
        this.form = false;
      }
    },
    async finishLater() {
      const result = await this.$swal({
        title: "Finalizar mais tarde",
        text:
          "Para resgatar esta operação acesse a aba Relatórios e busque pelo status Simulação.\n Lembre-se, a aprovação tem validade de sete dias corridos a contar da data de hoje.",
        icon: "info",
        confirmButtonText: "Acessar Relátorios",
        cancelButtonText: "Fechar",
        showCancelButton: true,
        showConfirmButton: true,
        customClass: "swal-listo",
      });

      if (result.value) {
        this.$router.push("/financiamentos/auto/relatorio");
      } else {
        this.closeSimulation();
      }
    },
    async isPinPadConnected() {
      try {
        await this.$services.pinpad.isPinpadConnected(this.affiliateId);
        this.$emit("continue");
      } catch (error) {
        console.error({ error });
        throw this.$swal({
          title: "Atenção!",
          text:
            "Sua maquininha (PinPad) não está conectada. Conecte-a agora ou resgate esta operação na sua página de Relatórios de financiamento",
          type: "warning",
          confirmButtonText: "OK",
          showCloseButton: true,
          customClass: "swal-listo",
        });
        this.finishLoad();
      }
    },

    async payNow() {
      if (!this.form && this.statusCar) {
        this.form = true;
        return false;
      }
      try {
        this.startLoad();
        await this.sendChassi();
        await this.isPinPadConnected()
      } catch (error) {
        this.form = false;
        this.finishLoad();
        console.error(error);
      }
    },
    async sendChassi() {
      let registerChassi = {
        registerLoanId: this.registerLoanId,
      };
      if (this.statusCar) {
        if (this.chassi.length <= 0) {
          return false;
        }
        registerChassi = {...registerChassi, chassi: this.chassi.toUpperCase().replace(/\s/g, ""),}
      }
      try {
        await this.createChassi(registerChassi);
      } catch (error) {
        console.error(error);        
        if (this.vehicle.status === 'new') {
          return
        }
        await this.deniedChassi(error)
        throw error
      }
    },
    async deniedChassi(error) {      
      const msgError = error.response.data.errors.map((el, index) => {
        return `${el}. `
      })
      this.form = false;
      this.chassi = ''
      this.finishLoad();
      const isNewCar = this.vehicle.status === 'new'
      const typeText = isNewCar ? 'Chassi' : 'Placa'
      const errorText = `${typeText} não encontrad${isNewCar? 'o': 'a'} na base de Índice Nacional (BIN). \n Cancele a operação${isNewCar ? ' ou mude o número do chassi.' : '.'} `;
      const swalOptions = {
        title: "Erro",
        text: msgError,
        icon: "error",
        cancelButtonText: "Cancelar",
        customClass: "swal-listo",
        showCancelButton: true,
        showConfirmButton: false
      }
      if (isNewCar) {
        swalOptions.confirmButtonText = "Editar"
        swalOptions.showConfirmButton = true
      }
      const result = await this.$swal(swalOptions);
      if (!result.value) {
        this.closeModal();
      }
    },
    closeModal() {
      this.closeSimulation();
      this.$emit("close");
      this.form = false;
    },
    delayTouch($v) {
      $v.$reset();

      if (touchMap.has($v)) {
        clearTimeout(touchMap.get($v));
      }

      touchMap.set($v, setTimeout($v.$touch, 2000));
    },
  },
  computed: {
    ...mapState({
      vehicle: state => state.simulacao.fipe.vehicle,
      iszerokm: state => state.simulacao.fipe.iszerokm
    }),
    statusCar() {
      let statusCar = this.vehicle.status || this.zeroKm;

      if (statusCar === "new") {
        statusCar = true;
      } else if (statusCar === "used") {
        statusCar = false;
      }

      return statusCar;
    },
    inputClass() {
      return {
        "area-chassi__button": true,
        "area-chassi__button--next": true,
        "is-disabled": this.$v.chassi.$invalid && this.form,
      };
    },
  },
  watch: {
    cet(newValue, value) {
      this.chassi = newValue.shortCet.chassi || "";
    },
  },
};
</script>

<style lang="less">
@import "~assets/vars.less";
@import "vue2-dropzone/dist/vue2Dropzone.min.css";
.cet {
  width: 60%;
  h3 {
    font-family: "newjunebold";
    font-size: 20px;
    color: @color1;
    font-weight: bold;
    line-height: 18px;
  }
  h4 {
    font-family: "newjuneregular";
    font-size: 18px;
    color: @color1;
    font-weight: bold;
    margin: 0 0 30px 0;
    line-height: 14px;
  }
  ul {
    list-style-type: none;
    padding: 0;
  }
  p {
    font-family: "newjuneregular";
    font-size: 16px;
    color: @color1;
    text-align: left;
    margin: 0;
  }
}

.modal-header {
  position: relative;
}

.area-chassi {
  width: 100%;
  max-width: 330px;
  flex-wrap: wrap;

  &,
  &__buttons {
    display: flex;
  }

  &__information,
  &__buttons {
    width: 100%;
  }

  &__label {
    margin: 0 0 3px 0;
  }

  &__buttons {
    margin-top: 15px;
    justify-content: space-between;
  }

  &__button {
    font-size: 14px;
    width: 48%;
    border: none;
    transition: background 0.5s ease-in-out;
  }

  &__error-mesage {
    color: red;
    display: block;
  }

  // TODO: Botao fechar modal - refatorar
  &__close {
    position: absolute;
    top: -10px;
    right: -10px;
    background: transparent !important;
    border: none;
    outline: none;
    color: #ccc !important;
    padding: 0;
    margin: 0;
    width: 1.2em;
    font-family: "newjunebook" !important;
    font-size: 14px;
    transition: color 0.1s ease-out;

    &:hover {
      color: #f27474 !important;
    }
  }

  //TODO: Essa area precisa ser apagada em uma refatoracao
  &--error {
    .base-input {
      border-color: red !important;
    }
  }

  .base-input {
    box-sizing: border-box;
    height: 35px !important;
    padding: 0 10px !important;
    margin: 0 !important;
    text-transform: uppercase;
  }

  .is-disabled {
    pointer-events: none;
    opacity: 0.5;
  }

  .area-chassi {
    &__button {
      margin: 0;
    }

    &__button {
      color: #fff;

      &--next {
        background-color: #27a114;

        &:hover {
          background-color: lighten(#27a114, 5%);
        }
      }

      &--close {
        &:hover {
          background-color: lighten(#000, 10%);
        }
      }
    }
  }
}
</style>
