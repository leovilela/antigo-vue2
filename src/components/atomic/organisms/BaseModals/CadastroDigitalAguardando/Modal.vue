<template>
<transition name="fade" >
  <Base-Modal center class="cadastro-digital" v-if="active">
    <template #header>
      <img src="~/assets/img/logo-listo.png" alt="" />
    </template>
    <template #body>
      <h3>Aguardando cadastro digital</h3>
      <p>Informe ao cliente que é necessário se cadastrar no aplicativo Listo, enviado por SMS para o número de celular informado por ele.</p>
      <p>A partir destas informações será possível oferecer melhores condições de financiamento.</p>
    </template>
    <template #footer>
      <Base-Button
        class="btn-modal-cancelar"
        aria-label="Fechar"
        @click="$emit('cancel')"
      >
        FECHAR
      </Base-Button>
    </template>
  </Base-Modal>
</transition>
</template>

<script>
export default {

  props: ["active"],

};
</script>

<style lang="less" scoped>
@import "~assets/vars.less";
  .cadastro-digital {
  h3 {
    font-family: 'newjunebold';
    font-size: 24px;
    color: @color1;
    font-weight: bold;
    margin-bottom: 12px;
  }
  p {
    font-family: 'newjuneregular';
    font-size: 14px;
    color: @color1;
    text-align: left;
  }
  .btn-modal-continuar {
    font-family: 'newjunebold';
    font-size: 14px;
    color: @color7;
    text-transform: uppercase;
  }
  .btn-modal-cancelar {
    font-family: 'newjunebold';
    font-size: 14px;
    width: auto;
    background: @color1;
    color: @color7;
    margin-top: 40px;
    padding: 0 50px;
    text-transform: uppercase;
  }
}
</style>
