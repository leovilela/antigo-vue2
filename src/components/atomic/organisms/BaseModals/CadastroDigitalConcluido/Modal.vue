<template>
  <Base-Modal center v-if="active" class="cadastro-digital">
    <template #header>
      <img src="~/assets/img/loading.gif" alt="" />
    </template>
    <template #body>
      <h3>Cadastro digital concluído.</h3>
      <p>Próximo passo: Escolha o veículo a ser financiado.</p>
    </template>
    <template #footer>
      <Base-Button
        class="btn-modal-continuar"
        aria-label="Botão Continuar"
        @click="$emit('continue')"
      >
        CONTINUAR
      </Base-Button>
    </template>
  </Base-Modal>
</template>

<script>
export default {
  props: ["active"],
};
</script>
<style lang="less" scoped>
  @import "~assets/vars.less";
  .cadastro-digital {
  h3 {
    font-family: 'newjunebold';
    font-size: 24px;
    color: @color1;
    font-weight: bold;
    margin-bottom: 12px;
  }
  p {
    font-family: 'newjuneregular';
    font-size: 14px;
    color: @color1;
    text-align: left;
  }
  .btn-modal-continuar {
    font-family: 'newjunebold';
    font-size: 14px;
    color: @color7;
    text-transform: uppercase;
  }
}
</style>
