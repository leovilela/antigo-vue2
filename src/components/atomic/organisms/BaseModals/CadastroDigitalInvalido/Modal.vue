<template>
<transition name="fade" >
  <Base-Modal center class="cadastro-digital" v-if="active">
    <template #header>
      <img src="~/assets/img/logo-listo.png" alt="" />
    </template>
    <template #body>
      <h3>Cadastro digital incompleto</h3>
      <p>Informe ao cliente que é necessário finalizar o cadastro no aplicativo Listo, enviado por SMS para o número de celular informado por ele.</p>
    </template>
    <template #footer>
      <Base-Button
        class="btn-modal-cancelar"
        aria-label="Cancelar"
        @click="$emit('cancel')"
      >
        FECHAR
      </Base-Button>
      <Base-Button
        class="btn-modal-cancelar"
        aria-label="Cancelar"
        @click="$emit('sendSMS')"
      >
        ENVIAR SMS
      </Base-Button>
    </template>
  </Base-Modal>
</transition>
</template>

<script>
export default {

  props: ["active"],

};
</script>

<style lang="less" scoped>
@import "~assets/vars.less";
  .cadastro-digital {
  h3 {
    font-family: 'newjunebold';
    font-size: 24px;
    color: @color1;
    font-weight: bold;
    margin-bottom: 12px;
  }
  p {
    font-family: 'newjuneregular';
    font-size: 14px;
    color: @color1;
    text-align: left;
  }
  .btn-modal-continuar {
    font-family: 'newjunebold';
    font-size: 14px;
    color: @color7;
    text-transform: uppercase;
  }
  .btn-modal-cancelar {
    font-family: 'newjunebold';
    font-size: 14px;
    width: auto;
    background: @color1;
    color: @color7;
    margin-top: 40px;
    padding: 0 50px;
    text-transform: uppercase;
  }
}
</style>
