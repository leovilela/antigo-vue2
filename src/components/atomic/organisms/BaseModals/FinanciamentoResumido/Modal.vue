<template>
      <Base-Modal v-if="active" class="formalizacao">
        <template #header>
          <img src="~/assets/img/logo-aperto-de-maos.png" alt="" />
        </template>
        <template #body>
          <h3>Financiamento concluído!</h3>
          <p>Parabéns! Transação efetuada com sucesso.</p>
          <p>Agora, formalize este financiamento.</p>
        </template>
        <template #footer>
          <Base-Button
            class="btn-modal-formalizacao"
            aria-label="Close modal"
            @click="$emit('formalization')"
          >
            FORMALIZAÇÃO
          </Base-Button>
        </template>
      </Base-Modal>
</template>

<script>
export default {
  props: {
    active: {
      type: Boolean,
      required: false,
      default: false,
    },
  },
};
</script>

<style lang="less">
@import "~assets/vars.less";
@import "vue2-dropzone/dist/vue2Dropzone.min.css";
.formalizacao {
  width: 60%;
  h3 {
    font-family: 'newjunebold';
    font-size: 20px;
    color: @color1;
    font-weight: bold;
    line-height: 40px;
  }
  p {
    font-family: 'newjuneregular';
    font-size: 16px;
    color: @color1;
    line-height: 28px;
    margin: 0;
    text-align: left;
  }
  .btn-modal-formalizacao {
    font-family: 'newjunebold';
    font-size: 14px;
    width: auto;
    background: @color1;
    color: @color7;
    margin-top: 40px;
    padding: 0 50px;
    text-transform: uppercase;
  }
  .btn-modal-fechar {
    font-family: 'newjunebold';
    font-size: 14px;
    width: auto;
    background: @color1;
    color: @color7;
    margin-top: 40px;
    padding: 0 50px;
    text-transform: uppercase;
  }
}

</style>
