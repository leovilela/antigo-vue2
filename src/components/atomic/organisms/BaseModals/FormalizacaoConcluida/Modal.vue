<template>
  <Base-Modal v-if="active" class="formalizacao">
    <template #header>
      <img src="~/assets/img/logo-aperto-de-maos.png" alt="" />
    </template>
    <template #body>
      <h3>Formalização concluída!</h3>
      <p>Parabéns! Documentação assinada pelo cliente.</p>
      <p>Processo encerrado!</p>
    </template>
    <template #footer>
      <Base-Button
        class="btn-modal-fechar"
        aria-label="Fechar"
        @click="$emit('close')"
      >
        FECHAR
      </Base-Button>
    </template>
  </Base-Modal>
</template>

<script>
export default {
  props: {
    active: {
      type: Boolean,
      default: false,
    },
  },
};
</script>

<style lang="less">
@import "~assets/vars.less";
@import "vue2-dropzone/dist/vue2Dropzone.min.css";
.formalizacao {
  width: 60%;
  h3 {
    font-family: 'newjunebold';
    font-size: 20px;
    color: @color1;
    font-weight: bold;
    line-height: 40px;
  }
  p {
    font-family: 'newjuneregular';
    font-size: 16px;
    color: @color1;
    line-height: 28px;
    margin: 0;
    text-align: left;
  }
  .btn-modal-formalizacao {
    font-family: 'newjunebold';
    font-size: 14px;
    width: auto;
    background: @color1;
    color: @color7;
    margin-top: 40px;
    padding: 0 50px;
    text-transform: uppercase;
  }
  .btn-modal-fechar {
    font-family: 'newjunebold';
    font-size: 14px;
    width: auto;
    background: @color1;
    color: @color7;
    margin-top: 40px;
    padding: 0 50px;
    text-transform: uppercase;
  }
}

</style>
