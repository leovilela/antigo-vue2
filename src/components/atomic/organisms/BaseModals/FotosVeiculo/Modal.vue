<template>
  <Base-Modal v-if="active" class="fotos-automovel">
    <template #header>
      <Base-Button @click="toCloseModal()" class="btn-modal-close"><i class="icon-cancel"></i></Base-Button>
    </template>
    <template #body>
      <div class="box-dropzone">
        <img src="~/assets/img/ico-foto.png" width="126" height="111px" alt="" />
        <h3 class="dropzone-custom-title">Fotos do Veículo</h3>
        <p class="dropzone-custom-subtitle">Envie as fotos seguindo o modelo abaixo:</p>
        <div class="dropzone-cars">
          <vue-dropzone id="dropzone" ref="onUploadPhotoDocument" :options="dropzoneOptions" :useCustomSlot="true" @vdropzone-file-added="onDocumentCar" @vdropzone-removed-file="onRemovedDocumentCar" @vdropzone-complete="afterComplete">
            <div class="dropzone-custom-content">
              <img src="~/assets/img/icon-document.png" width="130" height="123" alt="" />
              <Base-Button class="btn-enviar-foto">{{ typeModal.labelDocument }}</Base-Button>
            </div>
          </vue-dropzone>
          <vue-dropzone id="dropzone" ref="onUploadPhotoFront" :options="dropzoneOptions" :useCustomSlot="true" @vdropzone-file-added="onPhotoFrontCar" @vdropzone-removed-file="onRemovedPhotoFrontCar" @vdropzone-complete="afterComplete">
            <div class="dropzone-custom-content">
              <img src="~/assets/img/ico-foto-carro-dianteira.png" width="192" height="123" alt="" />
              <Base-Button class="btn-enviar-foto">Foto Perfil</Base-Button>
            </div>
          </vue-dropzone>
          <vue-dropzone id="dropzone" ref="onUploadPhotoBack" :options="dropzoneOptions" :useCustomSlot="true" @vdropzone-file-added="onPhotoBackCar" @vdropzone-removed-file="onRemovedPhotoBackCar" @vdropzone-complete="afterComplete">
            <div class="dropzone-custom-content">
              <img src="~/assets/img/ico-foto-carro-traseira.png" width="192" height="123" alt="" />
              <Base-Button class="btn-enviar-foto">Foto Traseira</Base-Button>
            </div>
          </vue-dropzone>
        </div>
      </div>
      <div class="error error-newcar" v-show="labelError !== false">{{ labelError }}</div>
      <div class="observacoes">
        <p><strong>OBSERVAÇÕES:</strong></p>
        <p>1. A placa do carro deve estar legível em ambas as fotos do veículo.</p>
        <p>2. As informações do documento do carro devem estar legíveis.</p>
      </div>
    </template>
    <template #footer>
      <Base-Button
        class="btn-modal-continuar"
        aria-label="Enviar Fotos"
        @click="processFilesReport()"
      >
        ENVIAR FOTOS
      </Base-Button>
    </template>
  </Base-Modal>
</template>

<script>
import vue2Dropzone from "vue2-dropzone";
import { mapActions, mapState, mapMutations } from "vuex";

export default {
  components: {
    vueDropzone: vue2Dropzone,
  },
  props: {
    active: {
      type: Boolean,
      default: false,
    },
    isNewCar: {
      type: Boolean,
      default: false,
    },
    contractId: {
      type: [String, Number],
    },
  },
  data() {
    return {
      dropzoneOptions: {
        url: "/",
        maxFilesize: 6, // MB
        thumbnailWidth: 200,
        maxFiles: 1,
        thumbnailMethod: "contain",
        addRemoveLinks: true,
        dictRemoveFile: "Remover arquivo",
        autoProcessQueue: true,
        acceptedFiles: "image/jpeg,image/png",
        dictInvalidFileType: "Você não pode carregar arquivos desse tipo.",
        method: "PUT"
      },
      files: {
        documentCar: null,
        photoFrontCar: null,
        photoBackCar: null,
      },
      client: {},
      labelError: false,
    };
  },
  methods: {
    ...mapActions({
      getDocumentType: "simulacao/relatorio/getDocumentType",
    }),
    ...mapMutations(["startLoading", "finishLoading"]),
    async loadDocumentsType() {
      try {
        await this.getDocumentType();
      } catch (error) {
        console.warn({ error });
      }
    },
    afterComplete(){
      console.info('upload ok');
    },
    async uploadFilesReport(file, type, contractId) {
      try {
        this.startLoading();
        await this.$services.upload.uploadFilesReport(file, type, contractId);
      } catch (error) {
        console.warn(error);
      }
      this.finishLoading();
    },
    onDocumentCar(file) {
      this.files.documentCar = file;
    },
    onPhotoFrontCar(file) {
      this.files.photoFrontCar = file;
    },
    onPhotoBackCar(file) {
      this.files.photoBackCar = file;
    },
    onRemovedDocumentCar() {
      this.files.documentCar = null;
    },
    onRemovedPhotoFrontCar() {
      this.files.photoFrontCar = null;
    },
    onRemovedPhotoBackCar() {
      this.files.photoBackCar = null;
    },
    async processFilesReport() {
      if (!this.isNewCar) {
        if (this.files.documentCar !== null && this.files.photoFrontCar !== null && this.files.photoBackCar !== null) {
          await this.uploadFilesReport(this.files.documentCar, 5, this.clientContractID);
          await this.uploadFilesReport(this.files.photoFrontCar, 6, this.clientContractID);
          await this.uploadFilesReport(this.files.photoBackCar, 7, this.clientContractID);
          this.$emit("toClose");
        } else {
          this.labelError = "As 3 fotos são obrigatórias";
        }
      }

      if (this.isNewCar) {
        if (this.files.documentCar !== null) {
          await this.uploadFilesReport(this.files.documentCar, 5, this.clientContractID);
          if (this.files.photoFrontCar !== null && this.files.photoBackCar !== null) {
            await this.uploadFilesReport(this.files.photoFrontCar, 6, this.clientContractID);
            await this.uploadFilesReport(this.files.photoBackCar, 7, this.clientContractID);
          }
          this.$emit("toClose");
        } else {
          this.labelError = "Foto da Nota Fiscal do Veículo é obrigatória";
        }
      }
    },
    toCloseModal() {
      this.$emit("toClose");
      this.labelError = false;
    },
  },
  mounted() {
    this.loadDocumentsType();
  },
  computed: {
    ...mapState({
      documentsType: state => state.simulacao.relatorio.documentsType,
    }),
    clientDocumentNumber() {
      return this.documentNumber;
    },
    clientContractID() {
      return this.contractId;
    },
    typeModal() {
      if (this.isNewCar) {
        return {
          labelDocument: "Nota fiscal do Veículo",
        };
      }
      return {
        labelDocument: "Documento do Veículo",
      };
    },
  },
};
</script>

<style lang="less" >
@import "~assets/vars.less";
@import "vue2-dropzone/dist/vue2Dropzone.min.css";
  .fotos-automovel {
  position: relative;
  .modal {
    width: 100%;
  }
  .vue-dropzone {
    width: auto;
    border: none;
  }

  .box-dropzone {
    align-items: center;
    flex-direction: column;
    display: flex;
  }

  #dropzone {
    width: 250px;
    background-color: @color2;
    color: @color5;
  }

  #dropzone .dz-preview .dz-details {
    color: @color2;
    background-color: @color1;
  }

  .dropzone-custom-title {
    font-family: 'newjunebold';
    font-size: @ftBigTitSize;
    margin: 20px 0 0 0;
  }

  .dropzone-custom-subtitle {
    font-family: 'newjuneregular';
    font-size: @ftSizeLarge;
    margin: 0 0 20px 0;
  }

  .dropzone-items {
    display: flex;
  }

  .dropzone-cars {
    justify-content: space-around;
    display: flex;
  }

  .dropzone .dz-message {
    margin: 0;
  }

  .observacoes {
    text-align: left;
  }

  .btn-enviar-foto {
    font-family: 'newjunebold';
    font-size: @ftSizeSmall;
    color: @color2;
    background: @color1;
    margin-top: 30px;
    text-transform: uppercase;
  }

  .btn-modal-close {
    width: auto;
    font-size: @ftBigTitSize;
    color: @color1;
    background: transparent;
    border: none;
    position: absolute;
    top: 0;
    right: 0;
  }
  .error {
    font-family: Arial, Helvetica, sans-serif;
    font-size: 16px;
    color: #856404;
    background-color: #fff3cd;
    border-color: #ffeeba;
    padding: 15px;
  }
  .vue-dropzone>.dz-preview .dz-error-message {
    margin-left: auto;
    margin-right: auto;
    left: 0;
    width: 100%;
    text-align: center;
    top: 0;
  }
}
</style>
