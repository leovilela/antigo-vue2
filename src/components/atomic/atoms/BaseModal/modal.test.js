import { shallowMount } from "@vue/test-utils";
import Modal from "../BaseModal/modal";

const factory = () => shallowMount(Modal, {
  stubs: ["Base-Modal"],
});

describe("Modal", () => {
  test("mounts properly", () => {
    const wrapper = factory();
    expect(wrapper.isVueInstance()).toBeTruthy();
  });

  test("renders properly", () => {
    const wrapper = factory();
    expect(wrapper.html()).toMatchSnapshot();
  });
});
