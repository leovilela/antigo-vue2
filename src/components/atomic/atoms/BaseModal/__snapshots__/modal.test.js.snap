// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal renders properly 1`] = `
<div>
  <div role="dialog" class="modal-backdrop"></div>
  <div class="modal-dialog">
    <div class="modal">
      <!---->
      <!---->
      <!---->
    </div>
  </div>
</div>
`;
